copt/constraint.py

```diff
@@ -20,52 +20,43 @@ def __call__(self, x):
         else:
             return np.infty
 
-    def prox(self, x, step_size):
+    def prox(self, x, step_size=None):
         return euclidean_proj_l1ball(x, self.alpha)
 
-    def lmo(self, u, x):
-        """Return s - x, s solving the linear problem
+    def lmo(self, u, x, active_set=None):
+        """Return (s - x, s_rep, None, 1), with s solving the linear problem
     max_{||s||_1 <= alpha} <u, s>
     """
         abs_u = np.abs(u)
         largest_coordinate = np.argmax(abs_u)
+        sign = np.sign(u[largest_coordinate])
 
         update_direction = -x.copy()
-        update_direction[largest_coordinate] += self.alpha * np.sign(
-            u[largest_coordinate]
-        )
+        update_direction[largest_coordinate] += self.alpha * sign
 
-        return update_direction, 1
+        # Only useful for active_set management in pairwise FW
+        fw_vertex_rep = (sign, largest_coordinate)
 
-    def lmo_pairwise(self, u, x):
-        abs_u = np.abs(u)
-        largest_coordinate = np.argmax(abs_u)
+        return update_direction, fw_vertex_rep, None, 1.
 
-        update_direction = np.zeros_like(u)
-        update_direction[largest_coordinate] += self.alpha * np.sign(
-            u[largest_coordinate]
-        )
+    def lmo_pairwise(self, u, x, active_set):
+        update_direction, fw_vertex_rep, _,  _ = self.lmo(u, x)
+        update_direction += x
 
-        u_active = -u * np.sign(x)
-        ma_u_active = ma.array(u_active, mask=(u_active == 0))
-        largest_active = np.argmax(ma_u_active)
-        if largest_active == largest_coordinate:
-            # .. if s and v are the same vertex ..
-            # .. take a FW step ..
-            update_direction -= x
-            max_step_size = 1.0
-        if u_active[largest_active] > 0:
-            update_direction[largest_active] -= self.alpha * np.sign(x[largest_active])
-            max_step_size = np.abs(x[largest_active]) / self.alpha
-        else:
-            # the zero vertex wins
-            max_step_size = max(self.alpha - np.sum(np.abs(x)), 0) / self.alpha
-            if max_step_size == 0:
-                # .. early termination ..
-                update_direction[:] = 0
+        def correlate(vertex_rep, u):
```

Comment:
could you add a docstring or a comment on what this function does? Also, since it's not a public API function, would be better to name it `_correlate`